### ISSUE TITLE
Auto generated URL of commit SHA references link to wrong address

### ISSUE THREAD

**Issue by liyuan-rey** [gogs/gogs#6300]
**Describe the bug**
Auto generated URL of commit SHA references in markdown link to wrong address

**Gogs version and commit**
Gogs  0.11.91.0811

**Git version**
Git 2.24.1.windows.2

**Operating system**
Windows 10 Home (Chinese) 19041.450

**Database**
Unknown

**To Reproduce**
1. Edit README.md file locally, add line with a valid commit SHA value, save file.
2. Commit and push README.md to repository
3. View the render page of README.md in Gogs
4. Click the link of commit SHA references, the browser jump to wrong 404 page

**Can you reproduce the bug at https://try.gogs.io?**
No try yet

**Expected behavior**
A link with correct URL to commit SHA references.

**Actual behavior**
A link with wrong URL, unnecessary 'src/develop' path in the URL.

**Screenshots**
N/A

**Additional context**
N/A

**Comment by unknwon** [comment]
Thanks for the feedback!

> **Can you reproduce the bug at [try.gogs.io](https://try.gogs.io)?**
> No try yet

Please do so.

**Comment by liyuan-rey** [comment]
emm.... try.gogs.io has the same issue.

**Comment by liyuan-rey** [comment]
i can't paste the screenshot, please visit https://try.gogs.io/liyuan/issues-6300

**Comment by liyuan-rey** [comment]
![Snipaste_2020-09-03_13-49-53](https://user-images.githubusercontent.com/10598397/92076295-e8adf200-edec-11ea-878c-f5971ec2a647.png)

**Comment by unknwon** [comment]
@liyuan-rey Thank you!

Note: The correct link should be:

```diff
-https://try.gogs.io/liyuan/issues-6300/src/master/commit/b5be65c1381188fb2f589504cafb115cdec6ab43
+https://try.gogs.io/liyuan/issues-6300/commit/b5be65c1381188fb2f589504cafb115cdec6ab43
```

**Comment by manfer** [comment]
https://github.com/gogs/gogs/blob/23ff182d1f8df2978785772bc58cf0ebfd2aeb0c/internal/route/repo/view.go#L85-L101

Not sure if changing line 90 to,

```go
				p = markup.Markdown(p, c.Repo.RepoLink, c.Repo.Repository.ComposeMetas())
```

is a proper solution or if it will have any undesirable effect somewhere else.

**Comment by unknwon** [comment]
@manfer Thanks for looking! I think relative links to a file will break, e.g. link to another Markdown, display an image.

Another option seems achievable would be adding something like `repoLink` in the map that is returned by `c.Repo.Repository.ComposeMetas()` (the docstring says it is for the purpose of external issue tracker but we can expand the use cases)?

**Comment by unknwon** [comment]
The patch will be included in 0.12.3.

**Comment by unknwon** [comment]
The 0.12.3 has been released.
